### PR TITLE
Rename `hearth-cognito` to `hearth-wasm`

### LINKED ISSUE #59: Reorganize Wasm-based crates
So I'd like to reuse and republish our Wasm abstraction and wrapper code. `hearth-wasm` and `hearth-macros` are directly connected to Hearth, so I'd like to rename them to something else. The new crate names should be like `something` and `something-macros`. Any ideas for a name for our standalone Wasm abstraction system?

After that, we can rename `hearth-cognito` to `hearth-wasm`.

### PR DESCRIPTION
**Pull request by marceline-cramer**
Closes #59.

Decided not to tackle #167 in this PR because this is complex enough as-is. However, I'll follow this up with that.

### CODE REVIEW

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ beginning with a capital letter.
 
 Example titles:
 - commit in `hearth-core`: `core: add ProcessFactoryImpl`
-- commit in `hearth-client`: `client: add cognito dep and WasmPlugin`
+- commit in `hearth-client`: `client: draw debug draw before terminal`
```
Comment (airidaceae) [inline CONTRIBUTORS.md:85]: good catch

File: crates/hearth-wasm/src/lib.rs
Comment (airidaceae) [inline crates/hearth-wasm/src/lib.rs:1]: The code that was previous cognito feels significantly different from the code that was previously hearth-wasm. I think we should split it up. Im ok with you deciding to tackle that in #167.

**Review by airidaceae — APPROVED** [review]
I like this. Make sure you update all issues that refer to this as cognito as cleanup once this is merged.

File: crates/hearth-wasm/src/lib.rs
Comment (marceline-cramer) [reply crates/hearth-wasm/src/lib.rs:1]: Can do. I think it makes sense to bundle it closely to the `*Abi` structs, but I'll get there when I get there.

Comment (airidaceae) [reply crates/hearth-wasm/src/lib.rs:1]: yup!

**Pull request merged**